packages/sanity/src/core/store/_legacy/document/utils/__test__/sequentializeEvents.test.ts

```diff
@@ -0,0 +1,94 @@
+/* eslint-disable no-nested-ternary */
+import {expect, test} from '@jest/globals'
+import {from, lastValueFrom} from 'rxjs'
+import {toArray} from 'rxjs/operators'
+
+import {type MutationPayload} from '../../buffered-doc'
+import {type ListenerEventWithSnapshot} from '../../getPairListener'
+import {type MutationEvent} from '../../types'
+import {sequentializeListenerEvents} from '../sequentializeListenerEvents'
+
+function mutationEvent({
+  previousRev,
+  resultRev,
+  mutations,
+}: {
+  previousRev: string
+  resultRev: string
+  mutations: MutationPayload[]
+}): MutationEvent {
+  return {
+    type: 'mutation',
+    documentId: 'test',
+    transactionId: resultRev,
+    effects: {revert: [], apply: []},
+    mutations,
+    previousRev: previousRev,
+    resultRev: resultRev,
+    transition: 'update',
+    transactionCurrentEvent: 1,
+    transactionTotalEvents: 1,
+    visibility: 'transaction',
+  }
+}
+
+test("it accumulates events that doesn't apply in a chain starting at the current revision", async () => {
```

Comment:
this is a good test.  may be worth adding a case for the deadline timeout

Reply:
Good call. Done